pkg/network/tracer/connection/kprobe/config.go

```diff
@@ -59,10 +59,8 @@ func enabledProbes(c *config.Config, runtimeTracer, coreTracer bool) (map[probes
 		enableProbe(enabled, probes.TCPClose)
 		enableProbe(enabled, probes.TCPCloseFlushReturn)
 		enableProbe(enabled, probes.TCPConnect)
-		if c.FailedConnectionsSupported() && (runtimeTracer || coreTracer) {
```

Comment:
Why are we removing this check?

Reply:
now we track PID for the whole lifecycle of the connection (we use the PID in finish connect even if failed conns are disabled). The alternative to doing this is checking if failed conns are enabled in `tcp_finish_connect`, and if not, deleting the map entry. but it will still leak a bit, and hopefully the map cleaner will be enough to keep it small in that case